app/common.py

```diff
@@ -1817,3 +1888,83 @@ def query_transcript(transcript):
 
     return results
+
+
+def query_molecular_consequences_by_variants(normalized_variant_list, feature_consequence_list, query):
+    variant_list = []
+    for item in normalized_variant_list:
+        if "GRCh37" in item:
+            variant_list.append(item["GRCh37"])
+        if "GRCh38" in item:
+            variant_list.append(item["GRCh38"])
+
+    pipeline_part = [{'$match': {'$expr': {'$and': [{'$or': [{'$eq': ['$variantID', '$$myvariant_id']}]}]}}},
+                     {'$addFields': {}}]
+
+    if feature_consequence_list != []:
+        pipeline_part.append({'$match': {'$or': []}})
+        or_query = []
+
+        for feature_consequence in feature_consequence_list:
+            if feature_consequence['isSystem']:
+                or_query.append({'$and': [{'featureConsequence.code': {'$eq': feature_consequence['feature_consequence']}}, {'featureConsequence.system': {'$eq': feature_consequence['system']}}]})
+            else:
+                or_query.append({'$or': [
+                    {'featureConsequence.code': {'$regex': ".*"+str(feature_consequence['feature_consequence']).replace('*', r'\*')+".*"}},
+                    {'featureConsequence.display': {'$regex': ".*"+str(feature_consequence['feature_consequence']).replace('*', r'\*')+".*"}}
+                ]})
+        pipeline_part[-1]['$match']['$or'] = or_query
+        pipeline_part.append({"$unwind": "$featureConsequence"})
+        pipeline_part.append({'$match': {'$or': or_query}})
+        pipeline_part.append({"$group": {
+            "patientID": {
+                "$first": "$$ROOT.patientID"
+            },
+            "variantID": {
+                "$first": "$$ROOT.variantID"
+            },
+            "transcriptRefSeq": {
+                "$first": "$$ROOT.transcriptRefSeq"
+            },
+            "MANE": {
+                "$first": "$$ROOT.MANE"
+            },
+            "source": {
+                "$first": "$$ROOT.source"
+            },
+            "cHGVS": {
+                "$first": "$$ROOT.cHGVS"
+            },
+            "pHGVS": {
+                "$first": "$$ROOT.pHGVS"
+            },
+            "featureConsequence": {
+                "$push": "$$ROOT.featureConsequence"
+            },
+            "impact": {
+                "$first": "$$ROOT.impact"
+            }
+        }})
+
+    query['SPDI'] = {'$in': variant_list}
+
+    query_string = [{'$match': query},
+                    {'$lookup': {'from': 'MolecConseq', 'let': {'myvariant_id': '$variantID'}, 'pipeline': pipeline_part,
+                                 'as': 'molecularConsequenceMatches'}},
+                    {'$addFields': {}},
+                    {'$match': {'molecularConsequenceMatches': {'$exists': True, '$not': {'$size': 0}}}}]
+
+    try:
+        results = variants_db.aggregate(query_string)
+        results = list(results)
+    except Exception as e:
+        print(f"{e}")
```

Comment:
Please use the same `print()` pattern that's already used throughout this file (smth like `print(f"DEBUG: Error{e} under query_molecular_consequences_by_variants query={query}")`)